util/fibers/detail/utils.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2023, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#pragma once
+
+#include <cstdint>
+
+namespace util {
+namespace fb2 {
+
+namespace detail {
+
+inline void CpuPause() {
+#if defined(__i386__) || defined(__amd64__)
+  __asm__ __volatile__("pause");
+#elif defined(__aarch64__)
+  /* Use an isb here as we've found it's much closer in duration to
+   * the x86 pause instruction vs. yield which is a nop and thus the
+   * loop count is lower and the interconnect gets a lot more traffic
+   * from loading the ticket above. */
+  __asm__ __volatile__("isb");
+#endif
+}
+
+class CycleClock {
+ public:
+  static uint64_t Now() {
+#if defined(__x86_64__)
+    uint64_t low, high;
+    __asm__ volatile("rdtsc" : "=a"(low), "=d"(high));
+    return (high << 32) | low;
+#elif defined(__aarch64__)
+    uint64_t val;
+    __asm__ volatile("mrs %0, cntvct_el0" : "=r"(val));
+    return val;
+#else
+#error "Unsupported architecture"
+#endif
```

Comment:
Do you think that good ol' `clock()` is that slow?

Reply:
this is our top right now. system call `clock` is relatively slow when we perform fiber context switches many millions time per second. [vdso] is the optimized version of reading the clock,
![image](https://github.com/romange/helio/assets/3674760/6ebc590b-47ed-4609-bbb8-02753995ec5d)

Reply:
with this change,

![image](https://github.com/romange/helio/assets/3674760/7ef1adb7-e4ab-48d5-b095-607fb4cb3ad3)

moreover if I look at the graph, I see that "FiberInterface::Preempt()" CPU usage was cut by more than half.